api/Stream.cpp

```diff
@@ -162,9 +163,9 @@ float Stream::parseFloat(LookaheadMode lookahead, char ignore)
 {
   bool isNegative = false;
   bool isFraction = false;
-  long value = 0;
+  double value = 0.0;
   int c;
-  float fraction = 1.0;
+  unsigned int digits_post_comma = 0;
```

Comment:
If `value` is a floating point number, there is no need for this extra variable. It could be folded into `value`.

Reply:
That's a good question here. What's more computing time expensive - `pow` or repeatedly doing a double multiplication?

Reply:
However, a further argument against `pow` is that there's a question of availability (and with what argument types) across all supported platforms.